src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -229,6 +228,67 @@ def get_custom_templatetags_libraries(self) -> list[str]:
             f"{prefix}.objects_api.json_tags",
         ]
 
+    def update_payment_status(self, submission: "Submission", options: dict) -> None:
+        config = ObjectsAPIConfig.get_solo()
+        assert isinstance(config, ObjectsAPIConfig)
+        config.apply_defaults_to(options)
+
+        if not options["payment_status_update_json"]:
+            logger.warning(
+                "Skipping payment status update because not template was configured."
+            )
+            return
+
+        context = {
+            "variables": get_variables_for_context(submission),
+            "payment": {
+                "completed": submission.payment_user_has_paid,
+                "amount": submission.payments.sum_amount(),
+                "public_order_ids": ",".join(
+                    submission.payments.get_completed_public_order_ids()
+                ),
+            },
+        }
+
+        # FIXME: replace with better suited alternative dealing with JSON specifically
+        updated_record_data = render_from_string(
+            options["payment_status_update_json"],
+            context=context,
+            disable_autoescape=True,
+            backend=openforms_backend,
+        )
+        self._check_size_rendered_json(updated_record_data)
+
+        with catch_json_decode_errors():
+            updated_record_data = json.loads(updated_record_data)
+
+        updated_object_data = {
+            "record": {
+                "data": {"payment": updated_record_data},
+                "startAt": get_today(),
+            },
+        }
+
+        object_url = submission.registration_result["url"]
+        with get_objects_client() as objects_client:
+            response = objects_client.patch(
+                url=object_url,
+                json=updated_object_data,
+                headers={"Content-Crs": "EPSG:4326"},
+            )
+            response.raise_for_status()
+
+    def _check_size_rendered_json(self, record_data: str) -> None:
+        if (
+            data_size := sys.getsizeof(record_data)
+        ) > settings.MAX_UNTRUSTED_JSON_PARSE_SIZE:
+            formatted_size = filesizeformat(data_size)
+            max_size = filesizeformat(settings.MAX_UNTRUSTED_JSON_PARSE_SIZE)
+            raise SuspiciousOperation(
+                f"Templated out content JSON exceeds the maximum size {max_size} ("
+                f"it is {formatted_size})."
+            )
```

Comment:
`self` is unused here and nothing about this implementation is really tied to the objects API (it's just currently the only place where we use it), so I'd opt to put this in a simple module-level utility function.

We can then bundle/move it together with the `render_to_json` (and make it part of that function maybe) when we get to it.